app/src/main/java/com/example/eric/pokedex/MainActivity.java

```diff
@@ -0,0 +1,290 @@
+package com.example.eric.pokedex;
+
+import android.app.SearchManager;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Color;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.SearchView;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.util.SparseBooleanArray;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+
+public class MainActivity extends AppCompatActivity implements SearchView.OnQueryTextListener{
+
+    RecyclerView recyclerView;
+    PokedexAdapter pokedexAdapter;
+    ArrayList<Pokedex.Pokemon> copyPokemons;
+    ArrayList<Pokedex.Pokemon> filteredModelListType = new ArrayList<>();
+    Pokedex pokedex;
+    int mSelectedItem;
+    boolean isGridView = false;
+    private List<Pokedex.Pokemon> mModels = new ArrayList<>();
+    private String[] typeArray = {"Clear Filter", "Normal", "Fire", "Water", "Fighting", "Flying", "Grass", "Poison", "Electric", "Ground", "Psychic", "Rock", "Ice", "Bug", "Dragon", "Ghost", "Dark", "Steel", "Fairy"};
+    private ListView mDrawerList;
+    private ArrayAdapter<String> mAdapter;
+    static int minHp = 0, minAttack = 0, minDefense = 0, random, code;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
+        Toolbar myToolbar = (Toolbar) findViewById(R.id.my_toolbar);
+        setSupportActionBar(myToolbar);
+
+        mDrawerList = (ListView) findViewById(R.id.navList);
+        mAdapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, typeArray);
+        mDrawerList.setAdapter(mAdapter);
```

Comment:
No idea this existed until looking at code, should have a toast or snackbar indicate it exists